File: core/address.go
package core

import "github.com/louisevanderlith/husk"

type Address struct {
	StreeNo     int
	Street      string `hsk:"size(80)"`
	UnitNo      string `hsk:"size(15)"`
	EstateName  string `hsk:"size(50)"`
	Suburb      string `hsk:"size(40)"`
	City        string `hsk:"size(40)"`
	Province    string `hsk:"size(30)"`
	PostalCode  string `hsk:"size(5)"`
	Coordinates string `hsk:"size(100)"`
	IsDelivery  bool
}

func (a Address) Valid() (bool, error) {
	return husk.ValidateStruct(&a)
}

File: lib/bodies/address.dart
class Address {
  final num streetNo;
  final String street;
  final String unitNo;
  final String estateName;
  final String suburb;
  final String city;
  final String province;
  final String postalCode;
  final String coordinates;
  final bool isDelivery;

  Address(
      this.streetNo,
      this.street,
      this.unitNo,
      this.estateName,
      this.suburb,
      this.city,
      this.province,
      this.postalCode,
      this.coordinates,
      this.isDelivery);

  Map<String, dynamic> toJson() {
    return {
      "StreetNo": this.streetNo,
      "Street": this.street,
      "UnitNo": this.unitNo,
      "EstateName": this.estateName,
      "Suburb": this.suburb,
      "City": this.city,
      "Province": this.province,
      "PostalCode": this.postalCode,
      "Coordinates": this.coordinates,
      "IsDelivery": this.isDelivery
    };
  }
}
